src/extensions.js

```diff
@@ -54,7 +54,12 @@ const extensionsCommands = () => {
 
     cy.contains('Upload Package File').click()
 
-    document.getElementById('legacy-uploader').classList.remove('hidden')
+    // Wait until the element is available
+    cy.get('#legacy-uploader').should('exist').then(($uploader) => {
+      if ($uploader) {
```

Comment:
Can it really be that it is here null? If yes, can you add a comment in which case?

Reply:
Yes, it is `null` in all cases, backward tested 15 times always the error `TypeError: Cannot read properties of null (reading 'classList')`:

- Intel macOS 14.5 Sonoma
  - Cypress 13.10.0, Node v20.11.1
    - Electron 118 (headless) with branches 4.4-dev, 5.1-dev, 5.2-dev and 6.0-dev
    - Firefox v126 (GUI) with branch 4.4-dev
- Ubuntu 24.04 Noble Numbat
  - Cypress 13.10.0, Node v18.19.1
    - Electron 118 (headless) with branches 4.4-dev, 5.1-dev, 5.2-dev and 6.0-dev
- Microsoft Windows 11 Pro
  - Cypress 13.10.0, Node v20.13.1
    - Electron 118 (headless) with branches 4.4-dev, 5.1-dev, 5.2-dev and 6.0-dev
    - Edge 125 (GUI) with branch 4.4-dev
    - Chrome 125 (GUI) with branch 4.4-dev

My favourite comment is the way it is:
```
// Wait until the element is available
```
Or I can offer:
```
// document.getElementById('legacy-uploader').classList.remove('hidden')
// Results in TypeError: Cannot read properties of null (reading 'classList')
// Therefore, wait until the element is available
```

Reply:
But then we can remove that code completely inside the then function?

Reply:
> But then we can remove that code completely inside the then function?

- no, the code inside then() is the original action and needed to remove the hidden class from #legacy-uploader to make it visible and usable
- the new inserted should('exist') ensures Cypress is waiting until the element with the ID 'legacy-uploader' exists in the DOM
- the backward tests have shown that the problem 'Cannot read properties of null' occurs everywhere where it was tested -> the fix is needed
- the code change is already in use in https://github.com/joomla/manual-examples/blob/main/tests/cypress/support/commands.js as `myInstallExtensionFromFileUpload` until new joomly-cypress NPM with this fix is available